controllers/NgoController.js

```diff
@@ -27,13 +27,17 @@ exports.NgoStore = [
   body("name", "Name must not be empty.").isLength({ min: 1 }).trim(),
   body("email", "Email must not be empty.").isLength({ min: 1 }).trim(),
   asyncH(async (req, res, next) => {
+    req.body.vcode = generateId();
```

Comment:
rather than adding to body, it's better to generate it directly below or store in a local const.

Reply:
By adding 'vcode' to the req.body our generic save function would be able to handle it.